scripts/build.py

```diff
@@ -514,36 +613,23 @@ def get_hashes_json_file_contents(file_hashes):
     return 'var hashes = JSON.parse(\'%s\');' % (hashes_json)
 
 
-def save_hashes_as_json(target_filepath, file_hashes):
-    """Save hashes in JS file containing JSON for files that
-    are to be interpolated in the frontend.
-
-    Args:
-        target_filepath: str. Path relative to /oppia directory defining
-            location where to save the JS file with hashes. The final location
-            path would also contain hash.
-        file_hashes: dict(str, str). Dictionary of file hashes.
-    """
-
-    with open(target_filepath, 'w') as f:
-        f.write(get_hashes_json_file_contents(file_hashes))
-
-    file_hash = generate_md5_hash(target_filepath)
-    relative_filepath = os.path.relpath(
-        target_filepath, os.path.join(os.path.curdir, 'build', 'assets'))
-    filepath_with_hash = _insert_hash(target_filepath, file_hash)
-    os.rename(target_filepath, filepath_with_hash)
-    file_hashes[relative_filepath] = file_hash
-
-
 def minify_func(source_path, target_path, file_hashes, filename):
+    """Call the appropriate functions to handle different types of file
+    formats.
+    """
     if filename.endswith('.html'):
-        process_html(source_path, target_path, file_hashes)
+        print 'Building %s' % source_path
+        with open(source_path, 'r+') as source_html_file:
+            with open(target_path, 'w+') as minified_html_file:
+                process_html(source_html_file, minified_html_file, file_hashes)
```

Comment:
I prefer to do it this way simply because of unit testing. 

I can now passing in a StringIO object during testing and read the output StringIO instead of having to `process_html` actually making changes, writing to an actual file and then read from that file. I then have to clean up the minified HTMl file which was what I was trying to avoid to begin with.

Reply:
Ah OK. That sounds fine to me, thanks for explaining.